.github/workflows/cli.yml

```diff
@@ -0,0 +1,57 @@
+name: CLI Build
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    name: Check and build codebase
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [14.x, 16.x, 18.x]
+        output-types: ['tsx', 'jsx', 'onepage']
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - uses: pnpm/action-setup@v2.2.2
+        with:
+          version: 7
+
+      - name: Get pnpm store directory
+        id: pnpm-cache
+        run: echo "::set-output name=pnpm_cache_dir::$(pnpm store path)"
+
+      - name: Setup pnpm cache
+        uses: actions/cache@v3
+        with:
+          path: ${{ steps.pnpm-cache.outputs.pnpm_cache_dir }}
+          key: ${{ runner.os }}-pnpm-store-${{ hashFiles('./pnpm-lock.yaml') }}
+          restore-keys: |
+            ${{ runner.os }}-pnpm-store-
+
+      - name: Install dependencies
+        run: pnpm install
+
+      - name: Build CLI
+        run: pnpm run build:cli
+
+      - name: Install spectacle
+        run: pnpm install spectacle
+
+      - name: create examples directory and move into it
+        run: mkdir cli-build-examples
+
+      - name: ${{ matrix.output-types }} example
+        run: spectacle -t ${{ matrix.output-types }} -n example-${{ matrix.output-types }}
```

Comment:
Do we also want to:

- Install the new project
- Build the deck

?

Reply:
This is definitely a wip 😅, I believe this command creates the new example project, but yes after my talk with @gksander Im sure we want to `install` and `start` and `test` the example projects. Im just trying to get this to run right now tbh - 🤔 Maybe the github action will still be picked up in a 'draft pr'

Reply:
Sorry to jump the gun on a WIP request. Good luck and carry-on!

Reply:
👍 - Yep looks like the new `workflows/cli.yml` does get pick up when in draft pr so Ill keep it here until its working!